neuraxle/hyperparams/distributions.py

```diff
@@ -1007,3 +1016,141 @@ def narrow_space_from_best_guess(self, best_guess, kept_space_ratio: float = 0.5
+
+            if np.isneginf(single_min):
+                single_min = None
+
+            if np.isposinf(single_max):
+                single_max = None
+
+            distribution_instance = distribution_class(mean, std, hard_clip_min=single_min, hard_clip_max=single_max)
+
+            if use_quantized_distributions:
+                distribution_instance = Quantized(distribution_instance)
+
+            distributions.append(distribution_instance)
+
+        return DistributionMixture(distributions, distribution_amplitudes)
+
+    def rvs(self) -> float:
+        """
+        Will return a float value drawned from the distribution mixture.
+
+        :return: a float.
+        """
+        # Draw from which distribution to draw first using the amplitude as probabilities on how to draw from each.
+        distribution_index = np.random.choice(len(self.distributions), p=self.distribution_amplitudes, replace=True)
+
+        # Draw from distribution.
+        return self.distributions[distribution_index].rvs()
+
+    def pdf(self, x) -> float:
+        """
+        Calculate the mixture probability distribution value at position `x`.
+        :param x: value where the probability distribution function is evaluated.
+        :return: value of the probability distribution function.
+        """
+        pdf_result = 0
+
+        for distribution_amplitude, distribution in zip(self.distribution_amplitudes, self.distributions):
+            pdf_result += (distribution_amplitude * distribution.pdf(x))
+
+        return pdf_result
+
+    def cdf(self, x) -> float:
+        """
+        Calculate the mixture cumulative distribution value at position `x`.
+        :param x: value where the cumulative distribution function is evaluated.
+        :return: value of the cumulative distribution function.
+        """
+        cdf_result = 0
+
+        for distribution_amplitude, distribution in zip(self.distribution_amplitudes, self.distributions):
+            cdf_result += (distribution_amplitude * distribution.cdf(x))
+
+        return cdf_result
+
+    def mean(self) -> float:
+        """
+        Calculate mean of the mixture.
+        :return: mean value of the mixtture.
+        """
+        # Since mean is linear, the mean is simply the weighted summed (using distribution amplitude) of each distribution mean.
+        mean_result = 0
+
+        for distribution_amplitude, distribution in zip(self.distribution_amplitudes, self.distributions):
+            mean_result += (distribution_amplitude * distribution.mean())
+
+        return mean_result
+
+    def var(self) -> float:
+        """
+        Calculate variance of the mixture.
+        :return: standard deviation value of the mixtture.
+        """
+        # Usually the var will be the sum(Variance) + 2 sum(Covariance).
+        # But since all distribution function are independents, the equation is
+        # Var(a * X0 + b * X1 + c * X2) = a^2 * Var(X0) + b^2 Var(X1)+ c^2 Var(X2)
```

Comment:
Ah, also, for part 1 of the code review: All those comments in the code would be pertinent to instead write in the docstring. This way the comments would be compiled and visible on the documentation website :) 

For symbols, you can escape code with two " ` " in a row like this as an example: 
```
        Usually the var will be the ``sum(Variance) + 2 sum(Covariance)``.
```